configuration/azure/appconfig/appconfig.go

```diff
@@ -284,9 +292,16 @@ func (r *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler
 }
 
 func (r *ConfigurationStore) getSentinelKeyFromMetadata(metadata map[string]string) string {
-	if s, ok := metadata["sentinelKey"]; ok && s != "" {
-		return s
+	type sentinelKeyMetadata = struct {
+		SentinelKey string `mapstructure:"sentinelKey"`
 	}
+	var sentinelKey sentinelKeyMetadata
+	kitmd.DecodeMetadata(metadata, &sentinelKey)
+
+	if sentinelKey.SentinelKey != "" {
```

Comment:
Don't need this check if you're returning the empty string anyway

Reply:
We'll clean this up when we make the component stable in the future. We are simply so late in the release process that I don't want to hold up the release further by seeking another approval (any code change cancels the existing approval unfortunately)